spring-batch-plus/src/integrationTest/java/com/navercorp/spring/batch/plus/step/adapter/ItemStreamFluxReaderProcessorIntegrationTest.java

```diff
@@ -0,0 +1,374 @@
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.core.step.builder.StepBuilder;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamWriter;
+import org.springframework.batch.support.transaction.ResourcelessTransactionManager;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+import org.springframework.lang.NonNull;
+import org.springframework.transaction.TransactionManager;
+import reactor.core.publisher.Flux;
+
+import javax.sql.DataSource;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.navercorp.spring.batch.plus.step.adapter.AdapterFactory.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings("unchecked")
+class ItemStreamFluxReaderProcessorIntegrationTest {
+
+	private static final int TEST_REPEAT_COUNT = 5;
+
+	private static final Logger logger = LoggerFactory.getLogger(
+		ItemStreamFluxReaderProcessorIntegrationTest.class);
+
+	private static int onOpenReadCallCount = 0;
+	private static int readContextCallCount = 0;
+	private static int onUpdateReadCallCount = 0;
+	private static int onCloseReadCallCount = 0;
+
+	private static int processCallCount = 0;
+
+	private static int itemCount = 0;
+	private static int chunkCount = 0;
+
+	@BeforeEach
+	void beforeEach() {
+		onOpenReadCallCount = 0;
+		readContextCallCount = 0;
+		onUpdateReadCallCount = 0;
+		onCloseReadCallCount = 0;
+
+		processCallCount = 0;
+
+		itemCount = ThreadLocalRandom.current().nextInt(10, 100);
+		chunkCount = ThreadLocalRandom.current().nextInt(1, 10);
+
+		logger.debug("itemCount: {}, chunkCount: {}", itemCount, chunkCount);
+	}
+
+	@RepeatedTest(TEST_REPEAT_COUNT)
+	void testReaderProcessor() throws Exception {
+		// given
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(TestConfiguration.class);
+		JobRepository jobRepository = context.getBean(JobRepository.class);
+		ItemStreamFluxReaderProcessor<Integer, Integer> testTasklet = context.getBean(
+			"testTasklet",
+			ItemStreamFluxReaderProcessor.class);
+		ItemStreamWriter<Integer> emptyItemStreamWriter = context.getBean(
+			"emptyItemStreamWriter",
+			ItemStreamWriter.class);
+		Job job = new JobBuilder("testJob", jobRepository)
+			.start(
+				new StepBuilder("testStep", jobRepository)
+					.<Integer, Integer>chunk(chunkCount, new ResourcelessTransactionManager())
+					.reader(itemStreamReader(testTasklet))
+					.processor(itemProcessor(testTasklet))
+					.writer(emptyItemStreamWriter)
```

Comment:
SimpleStepBuilder requires both reader and writer, I added no operation writer in integration tests.
![image](https://github.com/user-attachments/assets/e6501adf-564f-49c2-8d60-a541d6ab6fb6)